adot-testbed/app/src/test/java/software/amazon/adot/testbed/LogsTests.java

```diff
@@ -0,0 +1,145 @@
+            .withCommand("--config", "/etc/collector/config.yaml", "--feature-gates=+adot.filelog.receiver,+adot.awscloudwatchlogs.exporter,+adot.file_storage.extension");
+
+        //Mount the log file for the file log receiver to parse
+        collector.withCopyFileToContainer(MountableFile.forClasspathResource(logFilePath), logFilePath );
+        if (LOCAL_CREDENTIALS != null && !LOCAL_CREDENTIALS.isEmpty()) {
+            collector.withCopyFileToContainer(MountableFile.forHostPath(LOCAL_CREDENTIALS), "/root/.aws/");
+        } else {
+            collector.withEnv(System.getenv());
+        }
+
+        collector.start();
+        collector.waitingFor(Wait.forHealthcheck());
+        return collector;
+    }
+
+    @Test
+    void testSyslog() throws Exception {
+        collector = createAndStartCollector("/configurations/config-rfcsyslog.yaml", "/logs/RFC5424.log");
+
+        validateLogs("logstream-rfcsyslog" , "/logs/RFC5424.log");
+        collector.stop();
+    }
+
+    @Test
+    void testLog4j() throws Exception {
+        collector = createAndStartCollector("/configurations/config-log4j.yaml", "/logs/log4j.log");
+
+        validateLogs("logstream-log4j" , "/logs/log4j.log");
+        collector.stop();
+    }
+
+    @Test
+    void testJson() throws Exception {
+        collector = createAndStartCollector("/configurations/config-json.yaml", "/logs/testingJSON.log");
+
+        validateLogs("logstream-json" , "/logs/testingJSON.log");
+        collector.stop();
+    }
+
+    void validateLogs(String logName, String logFilePath) throws Exception {
+        var file = new File(logFilePath);
+        var lines = new HashSet<String>();
+
+        try (InputStream inputStream = getClass().getResourceAsStream(logFilePath);
+             BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                lines.add(line);
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Error reading from the file: " + logFilePath, e);
+        }
+
+        var cwClient = CloudWatchLogsClient.builder()
+            .build();
+
+        RetryerBuilder.<Void>newBuilder()
+            .retryIfException()
+            .retryIfRuntimeException()
+            .retryIfExceptionOfType(org.opentest4j.AssertionFailedError.class)
+            .withWaitStrategy(WaitStrategies.fixedWait(10, TimeUnit.SECONDS))
+            .withStopStrategy(StopStrategies.stopAfterAttempt(5))
+            .build()
+            .call(() -> {
+                var now = Instant.now();
+                var start = now.minus(Duration.ofMinutes(2));
+                var end = now.plus(Duration.ofMinutes(2));
+                var response = cwClient.getLogEvents(GetLogEventsRequest.builder().logGroupName("/test/logs")
+                    .logStreamName(logName)
```

Comment:
There is a very tight coupling in the validation between the log file name and the stream it expects it in. We should reduce this coupling or clearly document it. 

For example: `The validation logic expects logs to be in the same log stream as the file name`. 

I think this is bad though. What happens if two instances of these test are ran at the same time. Both will write to the same log stream and log group. This could cause transient failures or even worse, false successes. Each test case should write to a log stream name that is unique to it's test case.

Reply:
acutally logName there meant to be log stream name. Sorry for the confusion changed the usage of variable name.